azurerm/internal/services/apimanagement/api_management_logger_resource.go

```diff
@@ -142,6 +148,9 @@ func resourceApiManagementLoggerCreate(d *schema.ResourceData, meta interface{})
 		parameters.Credentials = expandApiManagementLoggerApplicationInsights(appInsightsRaw)
 	}
 
+	resourceId := utils.String(d.Get("resource_id").(string))
+	parameters.ResourceID = resourceId
```

Comment:
since this is optional we may need to make this conditional:

```suggestion
	if resourceId := d.Get("resource_id").(string); resourceId != "" {
		parameters.ResourceID = utils.String(resourceId)
	}
```